pkg/storage/secret/metadata/decrypt_store_test.go

```diff
@@ -0,0 +1,124 @@
+package metadata
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-jose/go-jose/v3/jwt"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/ini.v1"
+
+	"github.com/grafana/authlib/authn"
+	"github.com/grafana/authlib/types"
+	secretv0alpha1 "github.com/grafana/grafana/pkg/apis/secret/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/infra/usagestats"
+	"github.com/grafana/grafana/pkg/registry/apis/secret/contracts"
+	encryptionmanager "github.com/grafana/grafana/pkg/registry/apis/secret/encryption/manager"
+	"github.com/grafana/grafana/pkg/registry/apis/secret/secretkeeper"
+	"github.com/grafana/grafana/pkg/registry/apis/secret/xkube"
+	encryptionprovider "github.com/grafana/grafana/pkg/services/encryption/provider"
+	encryptionservice "github.com/grafana/grafana/pkg/services/encryption/service"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/kmsproviders/osskmsproviders"
+	"github.com/grafana/grafana/pkg/setting"
+	encryptionstorage "github.com/grafana/grafana/pkg/storage/secret/encryption"
+)
+
+func Test_DecryptStore_DecryptFromKeeper(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	s := setupDecryptTestService(t).(*decryptStorage)
+
+	t.Run("decrypt an existent secure value returns no error", func(t *testing.T) {
+		exposedValue, err := s.decryptFromKeeper(ctx, xkube.Namespace("default"), "sv-test")
```

Comment:
Since we are setting up the tests with auth and everything, could this just test `s.Decrypt` instead?